src/requests/index.ts

```diff
@@ -41,12 +67,18 @@ export class CoreDomainClient {
         "authorizing a session with an admin client is not allowed"
       );
     }
-    this.client = graphQLClient({
+    const client = graphQLClient({
       url: this.url,
       authorizedSession: sessionDetails,
     });
+    this.setClient(client);
   }
 
+  // TODO: refactor below into client namespaces (e.g. SessionClient)
+  // namspacing prevents this class from growing too unwieldy while still allowing for
+  // a simple interface for callers.
+  // Namespacing also allows for terser function names (e.g. `client.session.lock("123")`)
```

Comment:
I suggest the new pattern of exporting a namespaced client means that we deprecate the existing pattern and stop adding to it.